### ISSUE TITLE
[BUG] Send inline images does not work for Google AI Studio

### ISSUE THREAD

**Issue by axel10** [SillyTavern/SillyTavern#3209]
### Environment

🪟 Windows

### System

Chrome 131.0.6778.140, Windows 11

### Version

1.12.9

### Desktop Information

v22.12.0 - release - gemini-1.5-flash-8b-exp-0924

### Describe the problem

I have checked Send inline images, but gemini does not receive the images, and I can see from the output log that no images are sent.

I have tried using other gemini models, but gemini cannot receive the image.

![image](https://github.com/user-attachments/assets/d95b116e-7695-49a3-bed0-2f178245fa15)

### Please tick the boxes

- [X] I have explained the issue clearly, and I included all relevant info
- [X] I have checked that this [issue hasn't already been raised](https://github.com/SillyTavern/SillyTavern/issues?q=is%3Aissue)
- [X] I have checked the [docs](https://docs.sillytavern.app/) ![important](https://img.shields.io/badge/Important!-F6094E)

**Comment by Cohee1207** [comment]
Any extensions being used?

**Comment by axel10** [comment]
No, I am not using any extensions. I have located the problem, which is located in line 445 of prompt-converters.js,

// merge consecutive messages with the same role   
if (index > 0 && message.role === contents[contents.length - 1].role) {    
  contents[contents.length - 1].parts[0].text += '\n\n' + parts[0].text;   
} else {   
     contents.push({   
        role: message.role,   
        parts: parts,   
    });   
}`

Since the image is located at position 2 or later in the parts array of role:user, the image is not added to contents.

I added the following two lines of code after line 445 to fix the problem.

const files = parts.slice(1)
contents[contents.length - 1].parts.push(...files)

**Comment by Cohee1207** [comment]
Covered in this commit for #3191:
https://github.com/SillyTavern/SillyTavern/pull/3191/commits/73614f2f8d97646a62f60859eefa8a01cb0de44f#diff-7deee360e6d785c8176e49f9007cf9764731b3939a63be3894c7631e3fd2c712R463-R470
